### PR TITLE
Decompile Most Of xBehaveGoalSimple.cpp

### PR DESCRIPTION
**Pull request by Gota7**
Decompilation of most of that file & friends.

### CODE REVIEW

File: asm/Game/zNPCGoalTiki.s [hunk 1]
```diff
@@ -434,6 +434,24 @@ __ct__16zNPCGoalTikiIdleFi:
...
+
+Clear__16zNPCGoalTikiHideFv:
+/* 800ED1D4 000E9FD4  4E 80 00 20 */	blr 
+
+Clear__18zNPCGoalTikiPatrolFv:
+/* 800ED1D8 000E9FD8  4E 80 00 20 */	blr 
+
+Clear__16zNPCGoalTikiIdleFv:
+/* 800ED1DC 000E9FDC  4E 80 00 20 */	blr 
+
```
Comment (mattbruv) [inline asm/Game/zNPCGoalTiki.s]: Why were these added here?

Comment (Gota7) [reply asm/Game/zNPCGoalTiki.s]: I don't know how that happened.

File: .clang-format [hunk 2]
```diff
@@ -20,7 +20,6 @@ BinPackArguments: true
 BinPackParameters: true
 BraceWrapping:
   AfterClass: true
-  AfterCaseLabel: true
```
Comment (mattbruv) [inline .clang-format]: Why change this?

File: asm/Core/x/xBehaveGoalSimple.s [hunk 3]
```diff
@@ -243,25 +91,15 @@ __ct__12xGoalGenericFi:
...
-/* 8010EEBC 0010BCBC  4E 80 00 20 */	blr 
-
-Clear__12xGoalGenericFv:
-/* 8010EEC0 0010BCC0  4E 80 00 20 */	blr 
-
 Name__10xGoalEmptyFv:
 /* 8010EEC4 0010BCC4  3C 60 80 27 */	lis r3, lbl_8026A3B0@ha
 /* 8010EEC8 0010BCC8  38 63 A3 B0 */	addi r3, r3, lbl_8026A3B0@l
 /* 8010EECC 0010BCCC  38 63 00 0D */	addi r3, r3, 0xd
 /* 8010EED0 0010BCD0  4E 80 00 20 */	blr
```
Comment (mattbruv) [inline asm/Core/x/xBehaveGoalSimple.s]: I feel like you should be able to do this little 4 liner ;)

File: asmdiff.sh [hunk 4]
```diff
@@ -3,4 +3,4 @@
 OBJDUMP="$DEVKITPPC/bin/powerpc-eabi-objdump -D -bbinary -EB -mpowerpc -M gekko --adjust-vma=0x80003200"
 !(test -f baserom.dump) && $OBJDUMP baserom.dol > baserom.dump
 !(test -f main.dump) && $OBJDUMP main.dol > main.dump
-diff -u --color=always baserom.dump main.dump | head -n 2000
\ No newline at end of file
+diff -u --color=always baserom.dump main.dump | head -n 3000
```
Comment (mattbruv) [inline asmdiff.sh]: why change this?

File: src/Core/x/xBehaveGoalSimple.cpp [hunk 5]
```diff
@@ -2,44 +2,143 @@
 
 #include <types.h>
 
-// func_8010EB94
-#pragma GLOBAL_ASM("asm/Core/x/xBehaveGoalSimple.s", "xGoalSimple_RegisterTypes__FP8xFactory")
+extern const char lbl_8026A3B0[];
```
Comment (mattbruv) [inline src/Core/x/xBehaveGoalSimple.cpp]: If this is the string data for this file you should rename the label to something like `xBehaveGoalSimple_strings`

**Review by mattbruv — CHANGES_REQUESTED** [review]

File: .clang-format (hunk 2 above)
Comment (Gota7) [reply .clang-format]: Forgot to remove it from the commit, I need that to get clang-format to work.

File: asmdiff.sh (hunk 4 above)
Comment (Gota7) [reply asmdiff.sh]: Personal preference; again, forgot to remove it from the commit.

File: src/Core/x/xBehaveGoalSimple.cpp (hunk 5 above)
Comment (Gota7) [reply src/Core/x/xBehaveGoalSimple.cpp]: Sounds good.

**Review by mattbruv — APPROVED** [review]
